plugin/dnstap/setup_test.go

```diff
@@ -135,3 +140,27 @@ func TestMultiDnstap(t *testing.T) {
 		t.Error("expected third plugin to be last, but Next is not nil")
 	}
 }
+
+func Test_parseMessageTypes(t *testing.T) {
+	tests := []struct {
+		in     string
+		expect uint64
+	}{
+		{in: "", expect: ^uint64(0)},
+		{in: "CLIENT_QUERY", expect: 1 << tap.Message_CLIENT_QUERY},
+		{
+			in:     "CLIENT_QUERY CLIENT_RESPONSE",
+			expect: (1 << tap.Message_CLIENT_QUERY) | (1 << tap.Message_CLIENT_RESPONSE),
```

Comment:
Here you calculate the expected value the same way as you calculate it in the function you are testing. What if this calculation is wrong? Your test will not be able to catch this. I would just hardcode the expected value here.

On the other hand, the values like `Message_CLIENT_QUERY` are controlled by external project. If we hardcode the expected value, one day this test may surprisingly fail due to updates in external package which is not good. Though, I think it's unlikely that they (github.com/dnstap/golang-dnstap) will changes the message type constants.

Reply:
https://github.com/coredns/coredns/pull/6643/commits/6e9188f5af01462bdecb3f393c065cbf30d51854